web/lib/domain-logic/flashcards.js

```diff
@@ -0,0 +1,140 @@
+import { useState, useEffect, useCallback } from 'react'
+import { supabase as sb } from './supabaseClient'
+import { useUser } from './auth'
+
+const TB = { FlashcardSets: 'flashcard_sets', Flashcards: 'flashcards' }
+const getConfig = (config = {}) => ({
+  count: 'exact',
+  returning: 'representation',
+  ...config,
+})
+
+/* eslint-disable camelcase */
+const FlashcardSets = {
+  get: ({ user_id }) =>
+    sb
+      .from(TB.FlashcardSets)
+      .select('id, name, flashcards(*)')
+      .eq('user_id', user_id),
+  getOne: ({ user_id, id }) =>
+    sb
+      .from(TB.FlashcardSets)
+      .select('id, name, flashcards (*)', getConfig())
+      .eq('id', id)
+      .eq('user_id', user_id)
+      .single(),
+  upsert: (set) => sb.from(TB.FlashcardSets).upsert(set, getConfig()),
+  delete: ({ id }) => sb.from(TB.FlashcardSets).delete().match({ id }),
+}
```

Comment:
`FlashcardSets.get` sounds like Java, why not individual functions like `getFlashcardSets()`? Can be easier to tree-shake/ move stuff around if needed.